### ISSUE TITLE
Impossible to sign in to personal account to create a publish profile

### ISSUE THREAD

**Issue by Kagigz** [microsoft/BotFramework-Composer#5393]
## Describe the bug

To publish the bot, I clicked on "Add new publish profile" in the configuration section. I selected the option to create new Azure resources which requires signing in.

First bug is that the input doesn't take "@" as input for some reason, so I had to copy/paste my email address to enter it. I'm using my personal email address (@hotmail.fr)
2nd and most important problem: I have a message saying "We couldn't find a work or school account with that email address." 
I do have an Azure Subscription linked to that email address so I don't understand what's going on.

## Version

v1.3.0
![image](https://user-images.githubusercontent.com/16519462/102726403-cea60a00-431e-11eb-9198-cfff88b0a0ae.png)

## OS

- [ ] macOS
- [x] Windows
- [ ] Ubuntu

## To Reproduce

1. Create a bot on Composer (I used the LUIS + QnA  template)
2. Try creating a new publish profile
3. Try signing in with a personal account

## Expected behavior

I would expect to be able to sign in with my personal account linked to an Azure subscription.

## Screenshot

![image](https://user-images.githubusercontent.com/16519462/102726455-4b38e880-431f-11eb-9eb6-08c90cc056dc.png)

**Comment by luhan2017** [comment]
@tonyanziano @carlosscastro any idea on this? does this relate to the first party app settings?

**Comment by tonyanziano** [comment]
@luhan2017 I just took a look at our settings and personal accounts are not enabled. I will submit a request for them to be enabled. I will update once the request has gone through.

**Comment by tonyanziano** [comment]
It looks like first party apps cannot change their supported account types once they have been registered. I'm going to need to talk to 1st party app support to figure out how we can resolve this.

**Comment by Kagigz** [comment]
Thanks for that! 
There's also the problem of not being able to enter a "@" in the input field, which requires having to copy/paste.

**Comment by luhan2017** [comment]
@Kagigz you can type any character except "@"? I can't reproduce this in my side. 

![image](https://user-images.githubusercontent.com/32191031/102837349-95f95400-4436-11eb-80ed-e3aad2293f9b.png)

**Comment by veroneze** [comment]
Any updates on this?

**Comment by MIkey007007** [comment]
Any update? It appears that this is a very common problem. Let's hope we can get some progress.
What is the point to use the composer if it wouldn't publish?

**Comment by tonyanziano** [comment]
Hello,

I'm trying to reach out to the internal First Party Apps team here at Microsoft to debug the issue, but I believe most of them are out of office until the new year.

I will update when I hear back from them.

@luhan2017 @benbrown is there a way users can publish manually using the CLI as a workaround?

**Comment by MIkey007007** [comment]
Thanks for the update!

**Comment by MIkey007007** [comment]
Any new update? This problem happens frequently and can easily be reproduced. I have seen several other persons posts the same problem (phrase differently).

**Comment by tonyanziano** [comment]
We are currently waiting on approval from a member of an external team.

**Comment by MIkey007007** [comment]
I don't understand. What does it mean?

**Comment by Kagigz** [comment]
@luhan2017 yep that's right, cannot enter a '@' character. I have a French keyboard btw.

**Comment by MoesLab** [comment]
It seems the product does not go through initial testing of basic functions before being introduced to the public.

**Comment by ryan-daly-AgFCS** [comment]
> @luhan2017 yep that's right, cannot enter a '@' character. I have a French keyboard btw.

This could be completely incorrect as I'm not an expert on international keyboards, but perhaps can you try a "Right-Alt 2" to access the @ symbol? Looks like this might be how to do it with older French-Canadian mappings (worth a try!!!):

![image](https://user-images.githubusercontent.com/55713746/104231407-87d69c00-5414-11eb-9693-9b596a50fa02.png)


http://www3.uakron.edu/modlang/french/keyboard.html

**Comment by cwhitten** [comment]
@carlosscastro can you please jump in and share updates with the folks on this thread experiencing the personal account issue?

**Comment by JohnRobertGreen** [comment]
Also have the personal account issue. Worked fine in 1.2.1, upgraded to 1.3 and now it doesn't work.

**Comment by MoesLab** [comment]
Any news from the external team member yet?

**Comment by MIkey007007** [comment]
Can we get a new update?  Can timetable? The same problem is still happening.

**Comment by cwhitten** [comment]
Hi everyone. We're very very close to having this addressed. The internal team that we were waiting on is actively fixing this for us. I expect this to be wrapped up in the next few business days. 

Thank you for your patience.

**Comment by MIkey007007** [comment]
@cwhitten:  

Really appreciate the update! I am looking forward to the fix. :-)

**Comment by Hacwikir** [comment]
Hello, do you have any update? I am having the same issue.
Is there any workaround?

![image](https://user-images.githubusercontent.com/22629805/105933645-b1dab180-6003-11eb-89cf-5fb43864400c.png)

**Comment by operatedcoin** [comment]
Flagging this issue again @cwhitten @carlosscastro - have been trying workarounds but no luck.

**Comment by dbfr1** [comment]
Hi, I have the same problem with deploying app to AZ. can't publish on my AZ account with working subscription because of type of account (not work or student). Bot composer v.1.3.0 on MacOS Mojave 10.14.6.

**Comment by MoesLab** [comment]
I was thinking about what could be the reason(s) for a personal account in comparison with a commercial account technically, so one can be used but the personal one cause problem, but I did not get anywhere with it, maybe somebody can shed some light in this matter...

**Comment by operatedcoin** [comment]
I'm a beginner working in the space, but found an initial workaround through this thread https://github.com/microsoft/BotFramework-Composer/issues/3313. I was able to use the CLI to generate a new publishing profile and sign into Azure that way. The tenant ID seemed to enable me to get access in a way my account wouldn't. All the resources were generated in my portal but then the process pushed me back into composer to click Publish Bot and I got stuck with the same Sign In portal glitch and haven't been able to detect another CLI workaround.

**Comment by carlosscastro** [comment]
Hello, we are working closely with the AAD team every day to get this unblocked. Still no exact ETA for the release but we are treating this as a P0. We'll update by end of this week or earlier if there is more clarity around dates. The workaround mentioned above by @operatedcoin could be a mitigation until this fix goes out: https://github.com/microsoft/BotFramework-Composer/issues/3313.

**Comment by ic202** [comment]
I have the same problem. Is there an installation guide with FTP? I tried to install directly from Visual Studio but unfortunately I don't know in what structure the directories are. Is there an instruction somewhere to install a complete bot framework composer via ftp?

**Comment by carlosscastro** [comment]
@ic202 for the FTP feel free to bring the question to stack overflow, I haven't tried that and I don't know of any docs.

**Comment by carlosscastro** [comment]
Update: Even though we have enabled MSA auth for our app, we're seeing issues in our testing. The oncall person for the AAD team is helping us figure those out. Will update as we get more news.

**Comment by mitrro** [comment]
The issue is still there...

**Comment by carlosscastro** [comment]
Update: Our application was enabled for MSA, and the 2 missing steps were enabling MSA access to graph and ARM apis, to create the bot application and azure resources respectively. The graph access was cleared and rolled out today, and we see it working in our test environment. However, we encountered a technical issue acquiring ARM tokens for ARM APIs. MSA accounts interact with Azure through an intermediate tenant associated to the MSA account, and there are some complications around that interaction.  We are working with partner teams and internall to resolve it.

Because of this technical issue, the MSA account login is still unresolved and the team is working on the technical details to access ARM resources. This remains a top priority for us and we are making every effort to unblock it as soon as possible.

**Comment by mitrro** [comment]
> Update: Our application was enabled for MSA, and the 2 missing steps were enabling MSA access to graph and ARM apis, to create the bot application and azure resources respectively. The graph access was cleared and rolled out today, and we see it working in our test environment. However, we encountered a technical issue acquiring ARM tokens for ARM APIs. MSA accounts interact with Azure through an intermediate tenant associated to the MSA account, and there are some complications around that interaction. We are working with partner teams and internall to resolve it.
> 
> Because of this technical issue, the MSA account login is still unresolved and the team is working on the technical details to access ARM resources. This remains a top priority for us and we are making every effort to unblock it as soon as possible.

Appreciate the update, Carlos!

**Comment by MIkey007007** [comment]
Any new update?

**Comment by ic202** [comment]
I had the same problem.
I solved it by adding one new user to Azure Active Directory and giving him administrative access to my Subscription. 
With this new user, I can install the application ( Composer 1.3.1) without any problems.

**Comment by MIkey007007** [comment]
Hello, any new update ??????

**Comment by MoesLab** [comment]
I have not tried anything recently.

**Comment by carlosscastro** [comment]
Update: We're still working on the ARM - MSA signin scenario. Some details: When you log in to the Azure Portal with your MSa account, the azure portal does a special sign in where there is a tenant @onmicrosoft.com behind your MSA user and that is the token that can be used to access ARM (and not the token acquired directly with the MSA account). Also, users may have multiple tenants for an MSA accounts, so that leads to tenant selection: picking the right default tenant and in the future allowing users to select their tenant. These are the challenges we are working on. @tonyanziano and I are in the last stages of building a v1 implementation of this and looking to test it within the next 36 hours. If we confirm that it satisfies the scenarios we need for MSA, then it will be a matter of coding it and releasing the code. If, on the other hand, we find a blocker, we'd need to keep working with the ARM and Azure Portal teams. We'll update in either case to share progress. Thanks all for your patience and your feedback, we're working hard to get this great feature ready and make a smooth experience with MSA accounts.

**Comment by tonyanziano** [comment]
Update:

We have a scenario working in which a user can sign in with their MSA / personal account to interact with their Azure resources, using Composer's app credentials. However, we cannot get this working inside Electron -- what Composer is built with -- yet due to a limitation in the authentication library we are using.

This means that we finally have everything configured properly to support MSA login, and just need some extra flexibility from the library.

We are currently working hard with the owner of the authentication library to push a change through that will enable our scenario and allow us to implement this login flow into Composer.

Thanks again for all your patience, and we will continue to update this thread as we get more information.

**Comment by MoesLab** [comment]
I have updated the Framework and try again with a personal account, still the same issue. I can not Create new Azure resources in Configure resources dialog box. I am paying a monthly fee to Microsoft to be able to use Azure apps and functions as they promised and we getting spoon feed for the solution of major bugs. this is very frustrating it has been a long time since this bug was discovered.

**Comment by fahadmajeedrana** [comment]
> Update:
> 
> We have a scenario working in which a user can sign in with their MSA / personal account to interact with their Azure resources, using Composer's app credentials. However, we cannot get this working inside Electron -- what Composer is built with -- yet due to a limitation in the authentication library we are using.
> 
> This means that we finally have everything configured properly to support MSA login, and just need some extra flexibility from the library.
> 
> We are currently working hard with the owner of the authentication library to push a change through that will enable our scenario and allow us to implement this login flow into Composer.
> 
> Thanks again for all your patience, and we will continue to update this thread as we get more information.

Thanks for the update team. Downgraded to 1.2.1 now which is still publishing using access token. Looking forward to the fix.

Cheers.

**Comment by tonyanziano** [comment]
Update:

We have been unblocked by the authentication library, and now we are actively working on implementing the working MSA auth flow in a feature branch. We hope to have this merged into the main branch and available in a nightly build by the end of the week.

We will continue to keep this thread updated.

**Comment by tonyanziano** [comment]
Update:

We have merged a branch into the `main` branch #5909 that exposes APIs to support personal login. We are now working on integrating these APIs into the current Azure publishing flow.

**Comment by 17xande** [comment]
> I had the same problem.
> I solved it by adding one new user to Azure Active Directory and giving him administrative access to my Subscription.
> With this new user, I can install the application ( Composer 1.3.1) without any problems.

Clever work around. This worked for me to for now.

**Comment by tonyanziano** [comment]
This should be in the [latest nightly build](https://github.com/microsoft/BotFramework-Composer-Nightlies/releases). Please try out the publish flow with this build and verify that you can now publish using your personal account.

**Comment by ryan-daly-AgFCS** [comment]
Hello, The Moody's request submitted was invalid due to an incorrect subject line. Please ensure the request's subject line follows the below naming convention. PP ID - ######### Thank you, MaX


Regards,

Ryan Daly, M.S.
Data Scientist | Data Insights
AgCountry Farm Credit Services
1900 44th St S, Fargo, ND  58108
P: 701.499.2676
M: 701.306.4121
Email: ryan.daly@agcountry.com<mailto:ryan.daly@agcountry.com>
LinkedIn: ryanmdaly<https://www.linkedin.com/in/ryanmdaly/>
[Description: cid:image001.jpg@01CC5E54.F950FD00]


CONFIDENTIALITY NOTICE: This communication is intended for the exclusive use of the addressee(s).  It may contain information that is privileged, confidential and/or protected from disclosure, and that may be subject to the Electronic Communications Privacy Act of 1986, as amended.  If you are not the intended recipient, any review, dissemination, distribution or copying of the communication is strictly prohibited.  Please notify me immediately by telephone and/or reply e-mail and destroy this message and attachments.
